image/image_linux.go

```diff
@@ -392,6 +409,18 @@ func (s *imageSeeder) start(db *asserts.Database, optSnaps []*seedwriter.Options
 	return s.w.Start(db, newFetcher)
 }
 
+func (s *imageSeeder) validateSnapArchs(snaps []*seedwriter.SeedSnap) error {
+	for _, sn := range snaps {
+		for _, a := range sn.Info.Architectures {
+			if a != "all" && a != s.architecture {
```

Comment:
Is this check actually correct in all cases? A snap might be a "fat" snap that supports architecture: "amd64,i386" but because this check fails if it find the first non-match would error here? Or am I missing something?

Reply:
You are absolutely correct!! This code is wrong. The logic should be reversed!

Reply:
Fixed

Reply:
@mvo5 thanks for catching this, I remember being a bit perplexed, but it seems I didn't look at this closely/carefully enough